### PR TITLE
Feat simple moveit controller manager

### PR DESCRIPTION
**Pull request by ipa-hsd**
Follow up of #1140.
Currently the moveit config package generated by the Setup Assistant was not ready to use because of the missing controller configurations on the MoveIt! side. The PR generates a default `controllers.yaml` file along and sets the default `SimpleMoveItControllerManager`, which can be used within MoveIt! to communicate with `ros_control` or with the robot driver.

- f968db15ac6600af400c2044fd1dc8d5eafc2b5c - Currently `moveit_setup_assistant` generates an empty [`moveit_controller_manager.launch.xml`](https://github.com/ros-planning/moveit/blob/melodic-devel/moveit_setup_assistant/templates/moveit_config_pkg_template/launch/moveit_controller_manager.launch.xml). I see this has been done in https://github.com/ros-planning/moveit/pull/1051, but it also does many other things like launching Gazebo.   

- c171dc9e531c5e8dd5f5c2fcd94ce7d03a22acc0 - Generates `controller.yaml` (MoveIt! specific, which would be loaded in the launch file above), with default `FollowJointTrajectory` action interface

- 19c13b7a48190260e188c2d4810271cbb1d8a722 - Introduced in https://github.com/ros-planning/moveit/pull/951, but this is specific to MoveIt!. The `controller_list` will be needed irrespective of `ros_control` or not.

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Document API changes relevant to the user in the moveit/MIGRATION.md notes
- [ ] Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Decide if this should be cherry-picked to other current ROS branches
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by ipa-hsd** [comment]
https://github.com/ros-planning/moveit/pull/1140

Hope this time the rebase went well :)

**Comment by ipa-hsd** [comment]
Looks like something went wrong again. I'll rectify this today.

File: moveit_setup_assistant/templates/moveit_config_pkg_template/launch/moveit_controller_manager.launch.xml [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+
+  <!-- Set the param that trajectory_execution_manager needs to find the controller plugin -->
   <arg name="moveit_controller_manager" default="moveit_simple_controller_manager/MoveItSimpleControllerManager" />
   <param name="moveit_controller_manager" value="$(arg moveit_controller_manager)"/>
 
-  <!-- loads ros_controllers to the param server -->
-  <rosparam file="$(find [GENERATED_PACKAGE_NAME])/config/ros_controllers.yaml"/>
+  <!-- Flag indicating whether the controller manager should be used or not -->
+  <arg name="use_controller_manager" default="true" />
+  <param name="use_controller_manager" value="$(arg use_controller_manager)" />
```
Comment (rhaschke) [inline moveit_setup_assistant/templates/moveit_config_pkg_template/launch/moveit_controller_manager.launch.xml]: What's about this parameter? The whole code base doesn't know about this!

**Comment by rhaschke** [comment]
> You did move the controllers_list from ros_controllers into a separate file controllers.yaml and sent it to the param server using moveit_controller_manager.launch.xml, correct?

In https://github.com/ros-planning/moveit/pull/1140#issuecomment-440346622 you agreed to this statement, but I really don't see the benefit for this yet. Why do we need two separate files `controllers.yaml` and `ros_controllers.yaml`.

**Comment by rhaschke** [comment]
@MohmadAyman, could you please explicitly state whether the two approaches to configure controllers (controllers.yaml and ros_controllers.yaml) could be unified? My impression is, they can.

**Comment by MohmadAyman** [comment]
To be more clear ,yes without the PR everything should work the same, 'controller_list' is already being sent to paramedics server from ros_controlllers.yaml, @ipa-hsd

**Comment by ipa-hsd** [comment]
> controller_list' is already being sent to paramedics server from ros_controlllers.yaml

Yes.

The rationale was that not every setup would be using `ros_control`. In that case, the `controller_list` would have to manually written, which would be used by custom drivers as well.

File: moveit_setup_assistant/templates/moveit_config_pkg_template/launch/moveit_controller_manager.launch.xml (hunk 1 above)
Comment (davetcoleman) [reply moveit_setup_assistant/templates/moveit_config_pkg_template/launch/moveit_controller_manager.launch.xml]: Agreed, i don't see this parameter used anywhere else in the code, so it should be removed.

**Comment by rhaschke** [comment]
> To be more clear ,yes without the PR everything should work the same, 'controller_list' is already being sent to paramedics server from ros_controlllers.yaml

> The rationale was that not every setup would be using ros_control.

I don't think the previous approach was limited to ros_control, was it @MohmadAyman?
@ipa-hsd, maybe you were confused by the file name? 

Instead of having two nearly identical approaches to generate the controller configuration for ros_control and SimpleControllerManager, I suggest utilizing the same code, with a checkbox/flag and some switches to choose the correct behavior for one or the other option.

**Comment by MohmadAyman** [comment]
> I don't think the previous approach was limited to ros_control, was it @MohmadAyman?

Correct.

> Instead of having two nearly identical approaches to generate the controller configuration for ros_control and SimpleControllerManager, I suggest utilizing the same code, with a checkbox/flag and some switches to choose the correct behavior for one or the other option.

+1

**Comment by rhaschke** [comment]
@ipa-hsd, what is the status of this PR? Did you tried @MohmadAyman's new controller config in MSA?
It should make this PR superfluous. Hence, I'm tempted to close this one.

**Comment by ipa-hsd** [comment]
> I don't think the previous approach was limited to ros_control, was it @MohmadAyman?
@ipa-hsd, maybe you were confused by the file name?

True, I thought it was meant only to be used with `ros_control`. In that case, you could close it.

**Pull request closed without merging**